<!doctype html>
<title>removing the candidate source, onerror</title>
<script src="/resources/testharness.js"></script>
<script src="/resources/testharnessreport.js"></script>
<div id=log></div>
<script>
var timeout;
var v;
function createSource(src) {
  var source = document.createElement('source');
  source.src = src;
  return source;
}
var t = async_test(function(t) {
  v = document.createElement('video');
  document.body.appendChild(v); // invokes resource selection
  v.appendChild(createSource('/delay/?type=html&delay=50&content-type=video%2Fogg'));
  v.firstChild.onerror = t.step_func(function() { clearTimeout(timeout); t.done(); });
});
</script>
<!-- now resource selection algorithm will continue its sync section -->
<!-- #a is candidate -->
<!-- pointer is between #a and the end of the list -->
<script>
t.step(function() {
  v.removeChild(v.firstChild); // tests that we fire 'error' on it despite being removed
  timeout = setTimeout(t.step_func(function() { assert_unreached(); }), 1000);
});
</script>